public/app/plugins/panel/nodeGraph/useZoom.ts

```diff
@@ -1,45 +1,54 @@
 import { useCallback, useEffect, useRef, useState } from 'react';
 
+import { ZoomMode } from './panelcfg.gen';
+
 const defaultOptions: Options = {
   stepDown: (s) => s / 1.5,
   stepUp: (s) => s * 1.5,
   min: 0.13,
   max: 2.25,
+  zoomMode: ZoomMode.Cooperative,
 };
 
 interface Options {
   /**
    * Allows you to specify how the step up will be handled so you can do fractional steps based on previous value.
    */
-  stepUp: (scale: number) => number;
-  stepDown: (scale: number) => number;
+  stepUp?: (scale: number) => number;
+  stepDown?: (scale: number) => number;
 
   /**
    * Set max and min values. If stepUp/down overshoots these bounds this will return min or max but internal scale value
    * will still be what ever the step functions returned last.
    */
   min?: number;
   max?: number;
+
+  /**
+   * Sets how to handle zoom events when user is interacting with the page
+   */
+  zoomMode?: ZoomMode;
 }
 
 /**
  * Keeps state and returns handlers that can be used to implement zooming functionality ideally by using it with
  * 'transform: scale'. It returns handler for manual buttons with zoom in/zoom out function and a ref that can be
  * used to zoom in/out with mouse wheel.
  */
-export function useZoom({ stepUp, stepDown, min, max } = defaultOptions) {
+export function useZoom(options: Options) {
+  const { stepUp, stepDown, min, max, zoomMode } = { ...defaultOptions, ...options };
   const ref = useRef<HTMLElement | null>(null);
   const [scale, setScale] = useState(1);
 
   const onStepUp = useCallback(() => {
     if (scale < (max ?? Infinity)) {
-      setScale(stepUp(scale));
+      setScale(stepUp!(scale));
```

Comment:
Maybe check that `stepUp` exists before calling `setScale`? Just to be safe

Reply:
Refactored it a bit to make sure `stepUp` and `stepDown` is never undefined/null